dev/com.ibm.ws.microprofile.reactive.messaging.kafka/src/com/ibm/ws/microprofile/reactive/messaging/kafka/PartitionTrackerFactory.java

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.microprofile.reactive.messaging.kafka;
+
+import java.time.Duration;
+import java.util.concurrent.ScheduledExecutorService;
+
+import com.ibm.ws.microprofile.reactive.messaging.kafka.adapter.KafkaAdapterFactory;
+import com.ibm.ws.microprofile.reactive.messaging.kafka.adapter.TopicPartition;
+
+/**
+ *
+ */
+public class PartitionTrackerFactory {
+
+    private KafkaAdapterFactory adapterFactory = null;
+    private ScheduledExecutorService executor = null;
+    private int commitBatchMaxElements = 500;
+    private Duration commitBatchMaxInterval = Duration.ofMillis(500);
+    private boolean manualCommit = false;
```

Comment:
Does `manualCommit = !autoCommit`? It's called autoCommit elsewhere, why the inversion?

Reply:
It made more sense when I was writing it, since the manual case is the one where we have to do stuff. I'll change it.